### ISSUE TITLE
Incorrect specs added to httpc:cookie_header/1

### ISSUE THREAD

**Issue by jesperes** [erlang/otp#6846]
**Describe the bug**
Commit 61080083e6bf978b851d6fa590c3832d9130802c added an incorrect (or incomplete) type specification for the function `httpc:cookie_header/1`, which limits is to returning lists of 2-tuples, but it also does return single 2-tuples (see e.g. `httpc_cookie:header/4`. The incorrect spec is:
```erlang
-spec cookie_header(Url) -> [HttpHeader] | {error, Reason} when
      Url        :: uri_string:uri_string(),
      HttpHeader :: { Field :: [byte()], Value :: binary() | iolist()},
      Reason     :: term().
cookie_header(Url) ->
    cookie_header(Url, default_profile()).
```

**To Reproduce**
Given the following module
```erlang
-module(foo).
-export([main/0]).
main() ->
  Url = "http://anything.com",
  httpc:request(post, {Url, [httpc:cookie_header(Url)]}, [], []).
```
Compile it and run dialyzer:
```bash
$ erlc +debug_info foo.erl
$ dialyzer --build_plt --apps kernel stdlib inets >/dev/null 2>&1 # ignore warnings when building plt
$ dialyzer foo.beam
```
This runs without errors on OTP 24.x and earlier, but triggers a huge and very unreadable dialyzer error in OTP 25. I will paste the full error below. As mentioned, the spec says that the function only returns a list of 2-tuples (or some error term), but the function may also return a plain 2-tuple:
```
> erl
Erlang/OTP 25 [erts-13.1.3] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit:ns]

Eshell V13.1.3  (abort with ^G)
1> application:start(inets).                 
ok
2> httpc:cookie_header("https://anything.com").
{"cookie",[]}
```

**Expected behavior**
No dialyzer errors should be found.

**Affected versions**
OTP 25.x. OTP 24.x and earlier versions are not affected.

**Comment by jesperes** [comment]
Full dialyzer error message on OTP 25:
```erlang
foo.erl:3:1: Function main/0 has no local return
foo.erl:5:23: The call httpc:request
         ('post',
          {[1..255, ...],
           [[{[byte()],
              binary() |
              maybe_improper_list(binary() |
                                  maybe_improper_list(any(),
                                                      binary() | []) |
                                  byte(),
                                  binary() | [])}] |
            {'error', _},
            ...]},
          [],
          []) will never return since the success typing is 
         ('delete' | 'get' | 'head' | 'options' | 'patch' | 'post' |
          'put' | 'trace',
          {binary() |
           maybe_improper_list(binary() |
                               maybe_improper_list(any(), binary() | []) |
                               byte(),
                               binary() | []),
           [{[any()],
             binary() | maybe_improper_list(any(), binary() | [])}]} |
          {binary() |
           maybe_improper_list(binary() |
                               maybe_improper_list(any(), binary() | []) |
                               byte(),
                               binary() | []),
           [{[any()],
             binary() | maybe_improper_list(any(), binary() | [])}],
           binary() |
           maybe_improper_list(binary() |
                               maybe_improper_list(any(), binary() | []) |
                               byte(),
                               binary() | []),
           binary() |
           maybe_improper_list(binary() |
                               maybe_improper_list(any(), binary() | []) |
                               byte(),
                               binary() | []) |
           {fun((_) -> 'eof' | {_, _, _}), _} |
           {'chunkify', fun((_) -> 'eof' | {_, _, _}), _}},
          [{'autoredirect' | 'connect_timeout' | 'proxy_auth' |
            'relaxed' | 'ssl' | 'timeout' | 'version',
            'false' | 'infinity' | 'true' |
            binary() |
            maybe_improper_list(any(), binary() | []) |
            non_neg_integer() |
            {[any()], [any()]}}],
          [{'body_format' | 'full_result' | 'headers_as_is' |
            'ipv6_host_with_brackets' | 'receiver' | 'socket_opts' |
            'stream' | 'sync',
            atom() |
            binary() |
            fun((_) -> any()) |
            pid() |
            [any()] |
            {'self', 'once'} |
            {atom(), atom(), [any()]}}]) -> 
          {'error', _} | {'ok', _} and the contract is 
          (Method, Request, HttpOptions, Options) ->
             {'ok', Result} | {'error', term()}
             when
                 Method ::
                     'head' | 'get' | 'put' | 'patch' | 'post' |
                     'trace' | 'options' | 'delete',
                 Request ::
                     {uri_string:uri_string(), [HttpHeader]} |
                     {uri_string:uri_string(),
                      [HttpHeader],
                      ContentType :: uri_string:uri_string(),
                      HttpBody},
                 HttpBody ::
                     iolist() |
                     binary() |
                     {fun((Accumulator :: term()) ->
                              'eof' |
                              {'ok', iolist(), Accumulator :: term()}),
                      Accumulator :: term()} |
                     {'chunkify',
                      fun((Accumulator :: term()) ->
                              'eof' |
                              {'ok', iolist(), Accumulator :: term()}),
                      Accumulator :: term()},
                 HttpOptions :: [HttpOption],
                 HttpOption ::
                     {'timeout', timeout()} |
                     {'connect_timeout', timeout()} |
                     {'ssl', [ssl:tls_option()]} |
                     {'autoredirect', boolean()} |
                     {'proxy_auth', {string(), string()}} |
                     {'version', HttpVersion} |
                     {'relaxed', boolean()},
                 Options :: [OptionRequest],
                 OptionRequest ::
                     {'sync', boolean()} |
                     {'stream', StreamTo} |
                     {'body_format', BodyFormat} |
                     {'full_result', boolean()} |
                     {'headers_as_is', boolean()} |
                     {'socket_opts', [SocketOpt]} |
                     {'receiver', Receiver} |
                     {'ipv6_host_with_brackets', boolean()},
                 StreamTo ::
                     'none' | 'self' |
                     {'self', 'once'} |
                     file:name_all(),
                 SocketOpt :: term(),
                 BodyFormat :: string() | binary() | atom(),
                 Receiver ::
                     pid() |
                     fun((term()) -> term()) |
                     {ReceiverModule :: atom(),
                      ReceiverFunction :: atom(),
                      ReceiverArgs :: [any()]},
                 Result ::
                     {StatusLine, [HttpHeader], HttpBodyResult} |
                     {StatusCode, HttpBodyResult} |
                     RequestId | 'saved_to_file',
                 StatusCode :: non_neg_integer(),
                 StatusLine :: {HttpVersion, StatusCode, string()},
                 HttpVersion :: uri_string:uri_string(),
                 HttpHeader ::
                     {Field :: [byte()], Value :: binary() | iolist()},
                 HttpBodyResult :: uri_string:uri_string() | binary(),
                 RequestId :: any()
```

**Comment by kikofernandez** [comment]
Fix on the way #6856, sorry for the mistake that I introduced.
